sandbox/lua/decoders/linux_netdev.lua

```
@@ -0,0 +1,171 @@
+    | name:"lo_receive_fifo" type:integer value:0
+    | name:"lo_receive_errs" type:integer value:0
+    | name:"eth0_transmit_carrier" type:integer value:0
+    | name:"lo_receive_packets" type:integer value:601
+    | name:"lo_receive_bytes" type:integer value:50278
+    | name:"eth0_transmit_colls" type:integer value:0
+    | name:"eth0_receive_compressed" type:integer value:0
+    | name:"eth0_receive_errs" type:integer value:0
+    | name:"eth0_receive_multicast" type:integer value:0
+    | name:"eth0_transmit_bytes" type:integer value:0
+
+--]]
+
+local table = require "table"
+local l = require "lpeg"
+l.locale(l)
+
+local payload_keep = read_config("payload_keep")
+
+local space = l.P' '^0
+local keyword = (l.lower + l.digit)^1
+local counter = l.Ct(l.Cg(l.digit^1 / tonumber, "value") * l.Cg(l.Cc'2', "value_type"))
+local header = l.Ct(
+               l.P'Inter-'
+             * l.P'|'
+             * space
+             * l.P'Receive'
+             * space
+             * l.P'|'
+             * space
+             * l.P'Transmit'
+             * l.P'\n'
+             -- second line
+             * space
+             * l.P'face'
+             * space
+             * l.P'|'
+             * space
+             * l.Ct((l.Cg(keyword) * space)^1)
+             * l.P'|'
+             * space
+             * l.Ct((l.Cg(keyword) * space)^1)
+             * l.P'\n'
+             )
+local line = l.Ct(
+             space
+           * l.Cg(keyword)
+           * l.P':'
+           * space
+           * l.Ct((l.Cg(counter) * space)^1)
+           * l.P'\n'
+           )
+
+local grammar = l.Ct(
+                header
+              * l.Ct(line^1)
+              )
+
+local field_names = nil
+
+local msg = {
+    Timestamp = nil,
+    Type = "stats.netstat",
+    Payload = nil,
+    Fields = nil
+}
+
+function process_message()
+    local payload = read_message("Payload")
+
+    local data = grammar:match(payload)
+
+    if not data then
+        return -1
```

You can return a string error message as the second return value here, would be nice for users to know that the input failed to match the grammar.